bin/functions

```diff
@@ -1,6 +1,42 @@
 #!/usr/bin/env bash
 # shellcheck source=/dev/null
-source "${ASDF_DIR:-$HOME/.asdf}/lib/utils.bash"
+
+### >>>>>>
+### copy from  https://github.com/asdf-vm/asdf/blob/master/lib/utils.bash
+## fix https://github.com/halcyon/asdf-java/issues/177
+## fix https://github.com/halcyon/asdf-java/issues/188
+get_asdf_config_value_from_file() {
```

Comment:
the functions `asdf_dir` and `find_file_upwards` should also be vendored as they're used in the other 2 functions

Reply:
thanks ❤️